### PR TITLE
Add Supabase examples for REST & Edge Functions

### PR DESCRIPTION
**Pull request by beriberikix**
Signed-off-by: Jonathan Beri <jberi@golioth.io>

### CODE REVIEW

File: docs/data-routing/5-examples/24-supabase.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+The first example is the simplest and uses the REST API. Assuming a database
+named `golioth_pipeline_basic` with the following columns:
+
+| id | created_at | temp | lat | long |
+| -- | ---------- | ---- | --- | ---- |
+
+Create the following Pipeline:
+
+<Pipeline link='https://console.golioth.dev/pipeline?name=Supabase%20Edge%20Function%20Example&pipeline=c3RlcHM6CiAgLSBuYW1lOiBzdGVwMAogICAgZGVzdGluYXRpb246CiAgICAgIHR5cGU6IHdlYmhvb2sKICAgICAgdmVyc2lvbjogdjEKICAgICAgcGFyYW1ldGVyczoKICAgICAgICB1cmw6IGh0dHBzOi8vcmdnY3Fvc2x0cW9wY3d0d3doa2Muc3VwYWJhc2UuY28vZnVuY3Rpb25zL3YxL2dvbGlvdGgtcGlwZWxpbmUKICAgICAgICBoZWFkZXJzOgogICAgICAgICAgQXV0aG9yaXphdGlvbjogJFNVUEFCQVNFX0FVVEhfSEVBREVSCiAgICAgICAgICBhcGlrZXk6ICRTVVBBQkFTRV9BUElfS0VZCg=='/>
```
Comment (hasheddan) [inline docs/data-routing/5-examples/24-supabase.md:22]:
```suggestion
<Pipeline link='https://console.golioth.io/pipeline?name=Supabase%20Edge%20Function%20Example&pipeline=c3RlcHM6CiAgLSBuYW1lOiBzdGVwMAogICAgZGVzdGluYXRpb246CiAgICAgIHR5cGU6IHdlYmhvb2sKICAgICAgdmVyc2lvbjogdjEKICAgICAgcGFyYW1ldGVyczoKICAgICAgICB1cmw6IGh0dHBzOi8vcmdnY3Fvc2x0cW9wY3d0d3doa2Muc3VwYWJhc2UuY28vZnVuY3Rpb25zL3YxL2dvbGlvdGgtcGlwZWxpbmUKICAgICAgICBoZWFkZXJzOgogICAgICAgICAgQXV0aG9yaXphdGlvbjogJFNVUEFCQVNFX0FVVEhfSEVBREVSCiAgICAgICAgICBhcGlrZXk6ICRTVVBBQkFTRV9BUElfS0VZCg=='/>
```

File: docs/data-routing/5-examples/24-supabase.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+Assuming a database named `golioth_pipeline_advanced` with the following
+columns:
+
+| id | created_at | ce-time | ce-subject | ce-type | ce-source | temp | lat | long |
+| -- | ---------- | ------- | ---------- | ------- | --------- | ---- | --- | ---- |
+
+Create the following Pipeline:
+
+<Pipeline link='https://console.golioth.dev/pipeline?name=Supabase%20Edge%20Function%20Example&pipeline=c3RlcHM6CiAgLSBuYW1lOiBzdGVwMAogICAgZGVzdGluYXRpb246CiAgICAgIHR5cGU6IHdlYmhvb2sKICAgICAgdmVyc2lvbjogdjEKICAgICAgcGFyYW1ldGVyczoKICAgICAgICB1cmw6IGh0dHBzOi8vcmdnY3Fvc2x0cW9wY3d0d3doa2Muc3VwYWJhc2UuY28vZnVuY3Rpb25zL3YxL2dvbGlvdGgtcGlwZWxpbmUKICAgICAgICBoZWFkZXJzOgogICAgICAgICAgQXV0aG9yaXphdGlvbjogJFNVUEFCQVNFX0FVVEhfSEVBREVSCiAgICAgICAgICBhcGlrZXk6ICRTVVBBQkFTRV9BUElfS0VZCg=='/>
```
Comment (hasheddan) [inline docs/data-routing/5-examples/24-supabase.md:43]:
```suggestion
<Pipeline link='https://console.golioth.io/pipeline?name=Supabase%20Edge%20Function%20Example&pipeline=c3RlcHM6CiAgLSBuYW1lOiBzdGVwMAogICAgZGVzdGluYXRpb246CiAgICAgIHR5cGU6IHdlYmhvb2sKICAgICAgdmVyc2lvbjogdjEKICAgICAgcGFyYW1ldGVyczoKICAgICAgICB1cmw6IGh0dHBzOi8vcmdnY3Fvc2x0cW9wY3d0d3doa2Muc3VwYWJhc2UuY28vZnVuY3Rpb25zL3YxL2dvbGlvdGgtcGlwZWxpbmUKICAgICAgICBoZWFkZXJzOgogICAgICAgICAgQXV0aG9yaXphdGlvbjogJFNVUEFCQVNFX0FVVEhfSEVBREVSCiAgICAgICAgICBhcGlrZXk6ICRTVVBBQkFTRV9BUElfS0VZCg=='/>
```

File: docs/data-routing/5-examples/24-supabase.md (hunk 1 above)
Comment (hasheddan) [reply docs/data-routing/5-examples/24-supabase.md:22]: Also name of this one is "Supabase Edge Function Example", which me might want to switch to "Supabase REST API Example" 👍🏻

File: docs/data-routing/5-examples/24-supabase.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Create two [secrets](/data-routing/secrets) based on the
+[Service Role Key](https://supabase.com/docs/guides/api/api-keys#the-servicerole-key).
+`$SUPABASE_KEY` should be the the Server Role Key while `$SUPABASE_SERVICE_KEY`
+should take the form of 'Bearer
+$Service_Role_Key'. For example, if the `Service Role Key` is `12345` than the `$SUPABASE_SERVICE_KEY`should be set to`Bearer
+12345`.
+
+The second example uses Edge Functions. While
```
Comment (hasheddan) [inline docs/data-routing/5-examples/24-supabase.md:31]:
```suggestion
The second example uses [Edge Functions](https://supabase.com/docs/guides/functions). While
```

File: docs/data-routing/5-examples/24-supabase.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import Pipeline from '@site/src/components/usethispipeline'
+
+[Supabase](https://supabase.com/) is an open-source backend-as-a-service
+platform designed to simplify the development of web and mobile applications. It
+provides developers with a suite of tools including a Postgres database,
+real-time subscriptions, authentication, storage, and serverless functions, all
+managed through a single interface. Supabase provides multiple ways to consume
+data from Golioth.
+
+The first example is the simplest and uses the REST API. Assuming a database
```
Comment (hasheddan) [inline docs/data-routing/5-examples/24-supabase.md:14]:
```suggestion
The first example is the simplest and uses the [REST API](https://supabase.com/docs/guides/api). Assuming a database
```

File: docs/data-routing/5-examples/24-supabase.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+Create the following Pipeline:
+
+<Pipeline link='https://console.golioth.dev/pipeline?name=Supabase%20Edge%20Function%20Example&pipeline=c3RlcHM6CiAgLSBuYW1lOiBzdGVwMAogICAgZGVzdGluYXRpb246CiAgICAgIHR5cGU6IHdlYmhvb2sKICAgICAgdmVyc2lvbjogdjEKICAgICAgcGFyYW1ldGVyczoKICAgICAgICB1cmw6IGh0dHBzOi8vcmdnY3Fvc2x0cW9wY3d0d3doa2Muc3VwYWJhc2UuY28vZnVuY3Rpb25zL3YxL2dvbGlvdGgtcGlwZWxpbmUKICAgICAgICBoZWFkZXJzOgogICAgICAgICAgQXV0aG9yaXphdGlvbjogJFNVUEFCQVNFX0FVVEhfSEVBREVSCiAgICAgICAgICBhcGlrZXk6ICRTVVBBQkFTRV9BUElfS0VZCg=='/>
+
+Reuse or create two [secrets](/data-routing/secrets) based on the
+[Service Role Key](https://supabase.com/docs/guides/api/api-keys#the-servicerole-key).
+`$SUPABASE_KEY` should be the the Server Role Key while `$SUPABASE_SERVICE_KEY`
+should take the form of 'Bearer
+$Service_Role_Key'. For example, if the `Service Role Key` is `12345` than the `$SUPABASE_SERVICE_KEY`should be set to`Bearer
+12345`.
```
Comment (hasheddan) [inline docs/data-routing/5-examples/24-supabase.md:50]: Do you think if we added "For both of the examples..." we could just include this paragraph once on this page?

**Comment by beriberikix** [comment]
Thanks for the feedback! I think I incorporated all your suggestions - PTAL!

**Review by hasheddan — APPROVED** [review]
This is awesome! Thanks @beriberikix!

**Pull request merged**